compiler/rustc_middle/src/thir/visit.rs

```diff
@@ -129,7 +129,7 @@ pub fn walk_expr<'thir, 'tcx: 'thir, V: Visitor<'thir, 'tcx>>(
                 visitor.visit_expr(&visitor.thir()[base.base]);
             }
         }
-        PlaceTypeAscription { source, user_ty: _ } | ValueTypeAscription { source, user_ty: _ } => {
+        PlaceTypeAscription { source, .. } | ValueTypeAscription { source, .. } => {
```

Comment:
Please undo this change. (note how other things in this match always include all fields)